tests/formats_test/disklabel_test.py

```diff
@@ -80,7 +80,7 @@ def test_platform_label_types(self, arch):
         arch.is_pmac.return_value = False
 
         arch.is_efi.return_value = True
-        self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt"])
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt", "msdos"])
```

Comment:
I think you need to use a mock for `arch.is_arm` to make this work, since on non-arm this test _should_ fail due to the return value being `["gpt"]`. I suggest testing `get_platform_label_types` for both possible values of `arch.is_arm`.

Reply:
@dwlehman technically the UEFI standard supports either gpt or msdos no matter what the architecture

Reply:
That's not as important as the fact that `get_platform_label_types` currently returns ['gpt'] for non-arm, non-aarch64 UEFI platforms. The unit tests need to be updated to take into account `arch.is_arm` since you added logic around it to `get_platform_label_types`. It passes right now only because `disklabel.arch.is_arm` is a `Mock` which by default returns something that evaluates to True, meaning that `test_platform_label_types` will behave as if the test platform is arm all the time.

Reply:
Here's what I have in mind:

```diff
diff --git a/tests/formats_test/disklabel_test.py b/tests/formats_test/disklabel_test.py
index 21c2a748..4060e731 100644
--- a/tests/formats_test/disklabel_test.py
+++ b/tests/formats_test/disklabel_test.py
@@ -70,6 +70,7 @@ class DiskLabelTestCase(unittest.TestCase):
 
         arch.is_s390.return_value = False
         arch.is_efi.return_value = False
+        arch.is_arm.return_value = False
         arch.is_aarch64.return_value = False
         arch.is_pmac.return_value = False
 
@@ -80,9 +81,16 @@ class DiskLabelTestCase(unittest.TestCase):
         arch.is_pmac.return_value = False
 
         arch.is_efi.return_value = True
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt"])
+        arch.is_arm.return_value = True
         self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt", "msdos"])
+        arch.is_arm.return_value = False
         arch.is_efi.return_value = False
 
+        arch.is_aarch64.return_value = True
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["msdos", "gpt"])
+        arch.is_aarch64.return_value = False
+
         arch.is_s390.return_value = True
         self.assertEqual(disklabel_class.get_platform_label_types(), ["msdos", "dasd"])
         arch.is_s390.return_value = False
```